packages/metrics/src/model/metrics-result.ts

```diff
@@ -23,4 +29,68 @@ export interface MetricsResult<TFile = FileUnderTestModel, TMetrics = Metrics> {
    * The actual metrics
    */
   metrics: TMetrics;
+
+  constructor(name: string, childResults: MetricsResult<TFile, TMetrics>[], metrics: TMetrics, file?: TFile) {
+    this.name = name;
+    this.childResults = childResults;
+    this.metrics = metrics;
+    this.file = file;
+  }
+
+  public updateParent(value?: MetricsResult<TFile, TMetrics>) {
+    this.parent = value;
+    this.childResults.forEach((result) => result.updateParent(this));
+  }
+
+  public updateAllMetrics() {
```

Comment:
As I mentioned in one of our meetings: we shouldnt' need this method. In fact, we shouldn't need any public `calculate` method at all. Instead, we could move all this calculation magic inside the class and only recalculate on the fly when the metrics are requested and they are dirty (`undefined`).

So we start out with `#metrics: TMetrics | undefined` and in the getter of `metrics()` calculate when they are undefined. If we later update coverage or killed by from a child, we walk up the files and mark them dirty (setting `#metrics` to `undefined`).

What do you think?

Reply:
I assume that we would mark the `#metrics` undefined when we update the state of a mutant too? How does it work in that case?

Should we create a release with the tag `realtime-beta` when this gets merged?

Reply:
If we feel confident enough to merge we should feel confident enough to release. No need for a beta release, we don't have a release process for beta's in mutation-testing elements anyway.